dremio_client/flight/__init__.py

```diff
@@ -40,17 +80,36 @@ def connect(
         :param tls_root_certs_filename: use ssl to connect with root certs from filename
         :return: arrow flight client
         """
+        
+        scheme = "grpc+tcp"
+        connection_args = {}
+        
         if tls_root_certs_filename:
-            with open(tls_root_certs_filename) as f:
-                tls_root_certs = f.read()
-            location = "grpc+tls://{}:{}".format(hostname, port)
-            c = flight.FlightClient(location, tls_root_certs=tls_root_certs)
+            with open(tls_root_certs_filename) as root_certs:
+                connection_args["tls_root_certs"] = root_certs.read()
+            scheme = "grpc+tls"
         else:
-            location = "grpc+tcp://{}:{}".format(hostname, port)
-            c = flight.FlightClient(location)
-        if username:
-            c.authenticate(HttpDremioClientAuthHandler(username, password if password else ""))
-        return c
+            # use default unencrypted TCP connection
+            pass
+        
+        # Two WLM settings can be provided upon initial authentication
+        # with the Dremio Server Flight Endpoint:
+        # - routing-tag
+        # - routing queue
+        
+        client_auth_middleware = DremioClientAuthMiddlewareFactory()
+        client = flight.FlightClient("{}://{}:{}".format(scheme, hostname, port),
+                                     middleware=[client_auth_middleware], **connection_args)
+        
+        if username and password:
+            encoded_credentials = base64.b64encode(b'' + username.encode() + b':' + password.encode())
+            initial_options = flight.FlightCallOptions(headers=[
+                (b'authorization', b'Basic ' + encoded_credentials),
+                (b'routing-tag', b'test-routing-tag'),
```

Comment:
These (tag and queue) shouldn't be added unless they are passed as parameters to connect

Given the extra state we have to pass around I think it might be better to make a wrapper class to encapsulate the state and hide the direct pyarrow interaction

Reply:
yes, make sense. let me remove the tag and queue first

Reply:
great. Do you wnat to do taht as a separate PR or in this one?

Reply:
I would prefer a new PR lol